sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -94,7 +94,7 @@ public boolean accepts(Class<?> extensionType) {
                         invocationHandler));
     }
 
-    private Map<Method, Handler> methodHandlersFor(Class<?> sqlObjectType) {
+    private Map<Method, Handler> methodHandlersFor(Class<?> sqlObjectType, ConfigRegistry instanceConfig) {
```

Comment:
I've had a few days to think about this, and I think option 1 is the most practical approach: go ahead and use a config property, and explicitly document that the setting is locked in for a sql object type the first time it is instantiated (or invoked, for on-demand).